### PR TITLE
More PotionEffectType API

### PR DESCRIPTION
**Pull request by Machine-Maker**

### CODE REVIEW

**Comment by Machine-Maker** [comment]
Don’t do anything with this, I thought I’d a better idea to get the attribute modifier eight the attribute enum and just get it from the map that way and throw an exception if it doesn’t exist for that effect.

**Comment by Machine-Maker** [comment]
Rebased for 1.17.1

**Comment by Machine-Maker** [comment]
Refactored this to use the same `PaperRegistry` system from #6400

File: patches/server/0747-More-PotionEffectType-API.patch [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
++        return fromNMS(handle.getCategory());
++    }
++
++    @Override
++    public String translationKey() {
++        return this.handle.getDescriptionId();
++    }
++
++    public static EffectCategory fromNMS(net.minecraft.world.effect.MobEffectCategory mobEffectInfo) {
++        switch (mobEffectInfo) {
```
Comment (kezz) [inline patches/server/0747-More-PotionEffectType-API.patch]: We're in the future, `return switch` :p

Comment (Machine-Maker) [reply patches/server/0747-More-PotionEffectType-API.patch]: Ah yes, changed.

File: patches/api/0330-More-PotionEffectType-API.patch [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
++import java.util.Iterator;
++
++public interface PaperRegistry<T extends Keyed> extends Registry<T> {
++
++    /**
++     * Potion effect types.
++     *
++     * @see PotionEffectType
++     */
++    Registry<PotionEffectType> POTION_EFFECT_TYPE = new Registry<PotionEffectType>() {
```
Comment (Machine-Maker) [inline patches/api/0330-More-PotionEffectType-API.patch]: Need to change this to create a new PaperRegistry, not bukkit’s.

**Comment by Machine-Maker** [comment]
certainly NOT

**Comment by Machine-Maker** [comment]
Rebased for 1.18.1

**Comment by Machine-Maker** [comment]
Rebased and removed the whole `PaperRegistry` thing

**Review by electronicboy — APPROVED** [review]

**Pull request merged**